lib/cosmos/dart/spec/dart/dart_decommutator_spec.rb

```diff
@@ -0,0 +1,129 @@
+
+      hs_packet = Cosmos::System.telemetry.packet("INST", "HEALTH_STATUS")
+      hs_packets = []
+      # Write three packets. The first packet is always SYSTEM META.
+      3.times do
+        hs_packet.received_time = Time.now
+        hs_packets << hs_packet.clone
+        writer.write(hs_packet)
+        sleep 0.01
+      end
+      writer.shutdown
+      sleep 0.1
+
+      PacketLogEntry.all.each do |ple|
+        expect(ple.decom_state).to eq PacketLogEntry::NOT_STARTED
+      end
+
+      thread = Thread.new do
+        decom = DartDecommutator.new
+        decom.run
+      end
+      sleep 1 # Allow the decommutator to work
+      thread.exit
+
+      PacketLogEntry.all.each do |ple|
+        expect(ple.decom_state).to eq PacketLogEntry::COMPLETE
+      end
+
+      system_target = Target.where("name = 'SYSTEM'").first
+      meta = Packet.where("target_id = ? AND name = 'META'", system_target.id).first
+      packet_config = PacketConfig.where("packet_id = ?", meta.id).first
+      model = writer.get_decom_table_model(packet_config.id, 0)
+
+      meta_packet = Cosmos::System.telemetry.packet("SYSTEM","META")
+      meta_item_names = meta_packet.sorted_items.collect {|item| item.name }
+      expect(model.count).to eq 1 # One SYSTEM META packet
+      meta_row = model.first
+      expect(meta_row.reduced_state).to eq DartDecommutator::READY_TO_REDUCE
+      # Grab all the iXX column names which hold the actual data values
+      model.column_names.select {|name| name =~ /^i\d+/}.each_with_index do |item, index|
+        # RECEIVED_COUNT doesn't match because we receive additional SYSTEM META packets
+        # during decommutation that don't match those when writing the packet log entries
+        next if meta_item_names[index] == 'RECEIVED_COUNT'
+        db_value = meta_row.send(item.intern)
+        check_float(db_value, meta_packet.read(meta_item_names[index]))
+      end
+
+      system_target = Target.where("name = 'INST'").first
+      hs = Packet.where("target_id = ? AND name = 'HEALTH_STATUS'", system_target.id).first
+      packet_config = PacketConfig.where("packet_id = ?", hs.id).first
+      model = writer.get_decom_table_model(packet_config.id, 0)
+      expect(model.count).to eq 3 # We wrote 3 INST HEALTH_STATUS packets
+      decom_column_names = model.column_names.select {|name| name =~ /^i\d+/}
+
+      hs_packets.each_with_index do |packet, packet_index|
+        row = model.find(packet_index + 1)
+        expect(row.reduced_state).to eq DartDecommutator::READY_TO_REDUCE
+        db_index = 0
+        packet.sorted_items.each do |item|
+          # RECEIVED_COUNT doesn't match because DB counts while packet says 0
+          # TEMP1MIN, TEMP1MAX, TEMP1MEAN don't match because db says -100 while packet says 0
+          # Calling packet.process in clone seems to make them match because the packet
+          # Processor gets set instead of being uninitialized
```

Comment:
@ryanatball Can you explain this comment block?

Reply:
We should probably not decommutate RECEIVED_COUNT

Reply:
We should probably not decommutate ProcessorConversion based derived items, (and ideally not even run the processors)